plugins/builds/triggers/helpers.js

```diff
@@ -1074,26 +1073,35 @@ function isStageSetup(jobName) {
 }
 
 /**
- * Check if the job is a stage job
+ * get the stage name of a job
  * @param  {String} jobName                 Job name
  * @param  {Object} workflowGraph           Workflow Graph
- * @return {Boolean}
+ * @return {String}                         Stage name
  */
-function isStageJob(workflowGraph, jobName) {
+function getStageName(workflowGraph, jobName) {
     const jobNode = workflowGraph.nodes.find(n => n.name === jobName);
 
-    return jobNode.stageName !== undefined;
+    return jobNode ? jobNode.stageName : null;
 }
 
 /**
- * Check if the current job is a stage setup and the next job is a non-setup stage job
+ * Check if the current job is a stage setup and the next job is a non-setup job in the same stage
  * @param {String} triggerJob                Current job
  * @param {String} startFrom                 Next job
  * @param {Object} workflowGraph             Workflow Graph
  * @return {Boolean}
  */
 function isStartFromMiddleOfStage(triggerJob, startFrom, workflowGraph) {
```

Comment:
Can we rename the function name and args/variables
`isStartFromMiddleOfStage` -> `isStartFromMiddleOfCurrentStage`
`triggerJob` -> `currentJobName`
and may be `startFrom` -> `eventStartFrom`